pkg/api/v1/clients/kube/resource_client_factory.go

```diff
@@ -64,137 +67,174 @@ type ResourceClientSharedInformerFactory struct {
 	lock          sync.Mutex
 	defaultResync time.Duration
 
-	informers map[reflect.Type]cache.SharedIndexInformer
-	// startedInformers is used for tracking which informers have been started.
+	informers map[reflect.Type]map[string]cache.SharedIndexInformer
```

Comment:
maybe consider converting this `map[reflect.Type]map[string]cache.SharedIndexInformer` to some kind of struct for easier management? it''s not terrible, but having a struct enclosing these maps with some `Add` `Remove` functions can simplify this a bit

Reply:
Makes a lot of sense, I'll do it